credentials/fetcher_default.go

```diff
@@ -200,6 +208,7 @@ func (s *FetcherDefault) resolveAll(done chan struct{}, errs chan error, locatio
 	var wg sync.WaitGroup
 
 	for _, l := range locations {
+		l := l
```

Comment:
I again think it was a bug to leave the loop capture out here, because we use that to start a go routine.

In short, we need to capture the loop variable if we are using something concurrent with it.